test/pylib/cpp/common_cpp_conftest.py

```diff
@@ -0,0 +1,158 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import inspect
+import os
+import re
+import subprocess
+from copy import copy
+from functools import cache
+from pathlib import Path, PosixPath
+from typing import Dict, List
+
+import yaml
+from _pytest.nodes import Collector
+
+from test.pylib.cpp.item import CppFile
+
+ALL_MODES = {
+    'debug': 'Debug',
+    'release': 'RelWithDebInfo',
+    'dev': 'Dev',
+    'sanitize': 'Sanitize',
+    'coverage': 'Coverage',
+}
+DEBUG_MODES = {
+    'debug': 'Debug',
+    'sanitize': 'Sanitize',
+}
+DEFAULT_ARGS = [
+    '--overprovisioned',
+    '--unsafe-bypass-fsync 1',
+    '--kernel-page-cache 1',
+    '--blocked-reactor-notify-ms 2000000',
+    '--collectd 0',
+    '--max-networking-io-control-blocks=100',
+]
+
+
+def get_disabled_tests(config: Dict, modes: [str]) -> dict[str, set[str]]:
+    """
+    Get the dict with disabled tests.
+    Pytest spawns one process, so all modes should be handled there instead one by one as test.py does.
+    """
+    disabled_tests = {}
+    for mode in modes:
+        # Skip tests disabled in suite.yaml
+        disabled_tests_for_mode = set(config.get('disable', []))
+        # Skip tests disabled in the specific mode.
+        disabled_tests_for_mode.update(config.get('skip_in_' + mode, []))
+        # If this mode is one of the debug modes, and there are
+        # tests disabled in a debug mode, add these tests to the skip list.
+        if mode in DEBUG_MODES:
+            disabled_tests_for_mode.update(config.get('skip_in_debug_modes', []))
+        # If a test is listed in run_in_<mode>, it should only be enabled in
+        # this mode. Tests not listed in any run_in_<mode> directive should
+        # run in all modes. Inverting this, we should disable all tests
+        # that are listed explicitly in some run_in_<m> where m != mode
+        #  This, of course, may create ambiguity with skip_* settings,
+        # since the priority of the two is undefined, but oh well.
+        run_in_m = set(config.get('run_in_' + mode, []))
+        for a in ALL_MODES:
+            if a == mode:
+                continue
+            skip_in_m = set(config.get('run_in_' + a, []))
+            disabled_tests_for_mode.update(skip_in_m - run_in_m)
+        disabled_tests[mode] = disabled_tests_for_mode
+    return disabled_tests
+
+
+def read_suite_config(directory: Path) -> Dict[str, str]:
+    """
+    Helper method that will return the configuration from the suite.yaml file
+    """
+    with open(directory / 'suite.yaml', 'r') as cfg_file:
+        cfg = yaml.safe_load(cfg_file.read())
+        if not isinstance(cfg, dict):
+            raise RuntimeError('Failed to load tests in {}: suite.yaml is empty')
```

Comment:
Looks like a template string. What to replace `{}` with?

Reply:
Yes, better use an f-string and something inside the `{}`.